File: src/P01.java
// https://leetcode.com/problems/two-sum
//
/*

 */

import java.util.Arrays;
import java.util.HashMap;

public class P01 {
    public static int[] twoSum(int[] nums, int target) {
        var numIndexMap = new HashMap<Integer, Integer>();
        for (var i = 0; i < nums.length; i++) {
            var complement = target - nums[i];
            if (numIndexMap.containsKey(complement)) {
                return new int[]{numIndexMap.get(complement), i};
            }
            numIndexMap.put(nums[i], i);
        }
        return null;
    }

    public static void main(String[] args) {
        System.out.println(Arrays.toString(twoSum(new int[]{2, 7, 11, 15}, 9)));
    }
}

File: P01.cs
namespace lc_blind75;

class P01
{
    /// <remarks>
    ///     https://leetcode.com/problems/two-sum
    /// </remarks>
    public int[] TwoSum(int[] nums, int target)
    {
        var numIndexMap = new Dictionary<int, int>();
        for (var i = 0; i < nums.Length; i++)
        {
            var a = target - nums[i];
            if (numIndexMap.TryGetValue(a, out var _))
            {
                return [numIndexMap[a], i];
            }
            numIndexMap[nums[i]] = i;
        }
        throw new Exception("No two sum solution");
    }
}
